app/controllers/frontend_controller.rb

```diff
@@ -110,7 +110,7 @@ def pick_buyer_or_provider_layout
   end
 
   def all_services
-    @api_selector_services = current_user.accessible_services.includes(:account).to_a if current_account.try!(:provider?)
+    @api_selector_services = current_user.accessible_services.includes(:account) if current_account.provider?
```

Comment:
https://github.com/3scale/porta/blob/master/app/lib/authenticated_system.rb#L237

Hmm probably use `site_account` then?

Reply:
@hallelujah Why? 
As explained in [here](https://github.com/3scale/porta/pull/31#discussion_r222608470), I did this change because at the beginning of this controller there is this:
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/app/controllers/frontend_controller.rb#L21
Which does:
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/app/lib/authenticated_system.rb#L115-L117
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/app/lib/authenticated_system.rb#L80-L84
So here we don't need to do `current_account.try!(:provider?)` because there is current_account for sure 😄

Reply:
Ok so I am clearly wrong 😂 Although I dunno why 🤔 
![image](https://user-images.githubusercontent.com/11318903/46686258-06ba4a00-cbf8-11e8-89c2-e63e65be6863.png)

Reply:
ok well I was thinking of `current_account` can be `nil`

For example:

https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/lib/developer_portal/app/controllers/developer_portal/login_controller.rb#L3-L4

inherits from

https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/lib/developer_portal/app/controllers/developer_portal/base_controller.rb#L2
which inherits from

https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/lib/developer_portal/app/controllers/developer_portal/application_controller.rb#L4

The first controller has 
https://github.com/3scale/porta/blob/7da01b64214109778b8030d770bde282f6c3aca3/lib/developer_portal/app/controllers/developer_portal/login_controller.rb#L4

site_account is never nil as it comes from the request.

Reply:
ahaaa got it 👍 Thanks! 👍 
Then `site_account` is not appropriate either because it is trying to check if the logged account is a provider or not. However, thinking about it, I think there is no need to do that check either, because if `current_user.accessible_services` returns services, it already means that the current_account is a provider, because buyers don't have services 😄 
I've checked this in the preview DB to be sure (because I see this method called `accessible_services` is in `buyer_methods`, which is very confusing and probably by mistake):
```
irb(main):012:0> Account.buyers.where.has { exists Service.selecting { id.count }.grouping { account_id }.when_having { id.count >= 1 }.where.has { account_id == BabySqueel[:accounts].id } }.count
   (6215.4ms)  SELECT COUNT(*) FROM `accounts` WHERE `accounts`.`provider` = 0 AND `accounts`.`buyer` = 1 AND (EXISTS(SELECT COUNT(`services`.`id`) FROM `services` WHERE `services`.`account_id` = `accounts`.`id` GROUP BY `services`.`account_id` HAVING COUNT(`services`.`id`) >= 1))
=> 0
```